assets/js/base/context/cart-checkout/payment-methods/payment-method-data-context.js

```diff
@@ -276,6 +276,23 @@ export const PaymentMethodDataProvider = ( { children } ) => {
 		}
 	}, [ checkoutIsIdle, currentStatus.isSuccessful ] );
 
+	// if checkout has an error and payment is not being made with a saved token
+	// and payment status is success, then let's sync payment status back to
+	// pristine.
+	useEffect( () => {
+		if (
+			checkoutHasError &&
+			currentStatus.isSuccessful &&
+			! paymentData.hasSavedToken
```

Comment:
I'm curious, why is this last condition necessary? What would happen if a saved credit card has expired and the user tries to pay with it so the checkout fails? Wouldn't that also trigger the bug?

Reply:
When a saved payment method token is used, the payment status is set to ‘success’ because all the processing happens server side (there is no client side processing). So in the case that a saved payment method was expired and caused an error, there would be an error message returned from the server and the user could then select another payment method. So the bug here doesn’t affect saved payment methods tokens because it has a slightly different flow.

The payment method status is used primarily for client side payment method processing.